plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -209,15 +210,15 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
                 .map(JdbcColumnHandle.class::cast)
                 .collect(toImmutableList());
 
-        Set<JdbcColumnHandle> newColumnSet = ImmutableSet.copyOf(newColumns);
-
         if (handle.getColumns().isPresent()) {
+            Set<JdbcColumnHandle> newColumnSet = ImmutableSet.copyOf(newColumns);
```

Comment:
move of `newColumnSet` declaration  belongs to `Include grouping sets in JdbcTableHandle column list` commit